### PR TITLE
Ported source code for CommunityToolkit.WinUI.Media

### PR DESCRIPTION
**Pull request by Arlodotexe**
This PR brings over the source code for `CommunityToolkit.WinUI.Media` for #35.

This includes:
- Support for both UWP and WinAppSDK where possible
- Added nullable type annotations. 
- A minor correction to nullable annotations in the Animations package.
- A new internal `ArgumentNullExceptionExtensions` polyfill (courtesy of @Sergio0694) to enable clean argument validation.

This does **not** include:
- Any samples, tests or documentation. These will be added in a future PR.
- Backdrop brush functionality on WinAppSDK (unsupported)
- Support for any Uno Platform. The required Win2d and Composition APIs are not available.

### CODE REVIEW

File: components/Media/src/Brushes/RadialGradientBrush.cs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+namespace CommunityToolkit.WinUI.Media;
+
+/// <summary>
+/// RadialGradientBrush - This GradientBrush defines its Gradient as an interpolation
+/// within an Ellipse.
+/// </summary>
+[Obsolete("Please migrate to the RadialGradientBrush control from WinUI, this control will be removed in a future release. https://aka.ms/winui")]
+[ContentProperty(Name = nameof(GradientStops))]
+public partial class RadialGradientBrush : CanvasBrushBase
```
Comment (michael-hawker) [inline components/Media/src/Brushes/RadialGradientBrush.cs:20]: We shouldn't bring RadialGradientBrush over as it's in WinUI now.

**Review by michael-hawker — COMMENTED** [review]
Thanks Arlo, were there any other infrastructure or changes made with the sync from @Sergio0694?

**Comment by Arlodotexe** [comment]
> Thanks Arlo, were there any other infrastructure or changes made with the sync from @Sergio0694?

Most of the differences are in WASDK support, e.g. no Backdrop brushes or APIs.

File: components/Media/src/CommunityToolkit.WinUI.Media.csproj [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<Project Sdk="MSBuild.Sdk.Extras/3.0.23">
+  <PropertyGroup>
+    <ToolkitComponentName>Media</ToolkitComponentName>
+    <Description>This package contains Media.</Description>
+    <Version>0.0.1</Version>
```
Comment (Arlodotexe) [inline components/Media/src/CommunityToolkit.WinUI.Media.csproj:5]:
```suggestion
    <Version>8.0.0-beta.1</Version>
```

**Comment by michael-hawker** [comment]
@Arlodotexe looks like Win2D doesn't copy on the Any CPU option:

```
6>C:\Users\.nuget\packages\microsoft.graphics.win2d\1.0.5.1\build\net5.0-windows10.0.17763.0\Microsoft.Graphics.Win2D.targets(24,5): warning : Microsoft.Graphics.Canvas.dll could not be copied because the AnyCPU platform is being used. Please specify a specific platform to copy this file.
6>    1 Warning(s)
```

**Comment by michael-hawker** [comment]
Also getting this trying to run after x64 WASDK on the component solution:

image

**Comment by michael-hawker** [comment]
Ah, maybe because there's literally no samples/docs?

File: components/Animations/src/Xaml/Abstract/Animation{TValue,TKeyFrame}.cs [hunk 3]
```diff
@@ -86,11 +86,13 @@ public IList<IKeyFrame<TKeyFrame>> KeyFrames
     /// <summary>
     /// Gets the explicit target for the animation. This is the primary target property that is animated.
     /// </summary>
-    protected abstract string ExplicitTarget { get; }
+    protected abstract string? ExplicitTarget { get; }
 
     /// <inheritdoc/>
     public override AnimationBuilder AppendToBuilder(AnimationBuilder builder, TimeSpan? delayHint, TimeSpan? durationHint, EasingType? easingTypeHint, EasingMode? easingModeHint)
     {
+        default(ArgumentNullException).ThrowIfNull(ExplicitTarget);
```
[the remark this answers is not in the document]
Comment (Arlodotexe) [reply components/Animations/src/Xaml/Abstract/Animation{TValue,TKeyFrame}.cs:94]: This is just normal parameter validation, ExplicitTarget must not be null for this method to function. There's some other XAML-based classes where we also do parameter validation. 

ExplicitTarget is being provided as either explicitly null in `ClipAnimation` or possibly null in all the `EffectAnimation`s.
The pattern is a polyfill for the C# 12 feature, and is functionally equivalent to `Guard.IsNotNull` CommunityToolkit.Diagnostics but without the extra dependency.

File: components/Extensions/src/Markup/FontIconExtension.cs [hunk 4]
```diff
@@ -21,8 +21,11 @@ public class FontIconExtension : TextIconExtension
     public FontFamily? FontFamily { get; set; }
 
     /// <inheritdoc/>
-    protected override object ProvideValue()
+    protected override object? ProvideValue()
```
[the remark this answers is not in the document]
Comment (Arlodotexe) [reply components/Extensions/src/Markup/FontIconExtension.cs:24]: Correct, though looking again I think this should be changed to use proper parameter validation on the `Glyph` property. Passing a null value as the `Glyph` for `FontIcon` results in an exception being thrown, using parameter validation would match previous behavior with an improved error message.

**Review by michael-hawker — APPROVED** [review]

**Pull request merged**